TMDbLib/Objects/General/WatchProvidersByRegion.cs

```diff
@@ -0,0 +1,145 @@
+﻿using Newtonsoft.Json;
+
+namespace TMDbLib.Objects.General
+{
+    public class WatchProvidersByRegion
```

Comment:
This is a great candidate for a dictionary. We can remove this type, and use `Dictionary<string, WatchProviders>` in its place.

Reply:
The only issue here is the API surface that becomes less readable. 

Potentially, `WatchProvidersByRegion` could inherit from `Dictionary<string, WatchProviders>`.. maybe.

Reply:
I thought about this as well. Although this resembles a dictionary, I decided to go with a class because the [TMDb docs](https://developers.themoviedb.org/3/movies/get-movie-watch-providers) define it as an object with a fixed set of fields. The fields seem to correspond to the regions found at the bottom of the [JustWatch page](https://www.justwatch.com).

I just want to give my reasoning, but I would not object to changing it to a dictionary if you still think it's better.

I can see one pro and one con:

- 👍 If more regions are added, no maintenance is required to support it here.
- 👎 As you mentioned, the API becomes less readable and the user needs to know which keys to use.

Reply:
I don't recall having country codes or similar as properties in the remainder of TMDbLib, so I'll go with the dictionary method. From the docs, I think that API was seconds away from being a list (like our other result container) with country code as a property. :P

Reply:
Yep, that would have made more sense! I looked around the other endpoints and also didn't see any comparable response structures.

Reply:
Converted to a dictionary.